vembrane/modules/sort.py

```diff
@@ -0,0 +1,138 @@
+from vembrane.representations import Annotation, SourceEnvironment
+
+
+def add_subcommand(subparsers):
+    parser = subparsers.add_parser(
+        "sort",
+        description="Sort VCF records by one or multiple Python expressions that "
+        "encode keys for the desired order. This feature loads the entire VCF file "
+        "into memory in order to maximize performance. It is thus meant to sort small, "
+        "already filtered VCF files, e.g. for prioritizing records for the human eye. "
+        "For large VCF files, the only relevant sorting is usually by position, "
+        "which is better done with e.g. bcftools (and usually the sorting that variant "
+        "callers output).",
+    )
+    parser.add_argument(
+        "vcf",
+        help="The VCF/BCF file containing the variants. If not specified, "
+        "reads from STDIN.",
+        nargs="?",
+        default="-",
+    )
+    parser.add_argument(
+        "expression",
+        type=check_expression,
+        help="Python expression (or tuple of expressions) returning orderable values "
+        "to sort the VCF records by (ascending, smallest values coming first). "
+        "If multiple expressions are provided as a tuple, they are prioritized from "
+        "left to right with lowest priority on the right. "
+        "NA/NaN values are sorted to the end.",
+    )
+    parser.add_argument(
+        "--output",
+        "-o",
+        default="-",
+        help="Output file, if not specified, output is written to STDOUT.",
+    )
+    parser.add_argument(
+        "--output-fmt",
+        "-O",
+        default="vcf",
+        choices=["vcf", "bcf", "uncompressed-bcf"],
+        help="Output format.",
+    )
+    add_common_arguments(parser)
+
+
+def execute(args) -> None:
+    overwrite_number = {
+        "INFO": dict(args.overwrite_number_info),
+        "FORMAT": dict(args.overwrite_number_format),
+    }
+    fmt = {"vcf": "", "bcf": "b", "uncompressed-bcf": "u"}[args.output_fmt]
+
+    try:
+        with create_reader(
+            args.vcf,
+            backend=args.backend,
+            overwrite_number=overwrite_number,
+        ) as reader:
+            # TODO: unify the reader/writer setup across subcommands
+            reader.header.add_generic("vembraneVersion", __version__)
+            reader.header.add_generic(
+                "vembraneCmd",
+                "vembrane "
+                + " ".join(
+                    normalize(arg) if " " in arg else arg for arg in sys.argv[1:]
+                ),
+            )
+
+            annotation = Annotation(args.annotation_key, reader.header)
+
+            # load records into memory
+            # since sorting by something else than the position is only reasonable
+            # for already small, filtered VCF files, this is not an issue in practice.
+            # TODO: auto-detect larger files and use a mergesort library for those?
+            records = list(enumerate(reader))
```

Comment:
Turn this into an issue so we don't lose track. Right now, most (all?) vembrane commands support streaming, and this being an exception is somewhat unexpected for the end-user (who may or may not read the disclaimer in the documentation / help)